src/main/java/com/pileproject/drivecommand/model/nxt/NxtMachineSimulator.java

```diff
@@ -0,0 +1,52 @@
+package com.pileproject.drivecommand.model.nxt;
+
+import com.pileproject.drivecommand.machine.device.DeviceType;
+import com.pileproject.drivecommand.machine.device.input.LineSensor;
+import com.pileproject.drivecommand.machine.device.input.SoundSensor;
+import com.pileproject.drivecommand.machine.device.input.TouchSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedLineSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedSoundSensor;
+import com.pileproject.drivecommand.machine.device.input.simulation.SimulatedTouchSensor;
+import com.pileproject.drivecommand.machine.device.output.Motor;
+import com.pileproject.drivecommand.machine.device.output.simulation.SimulatedMotor;
+import com.pileproject.drivecommand.machine.device.port.InputPort;
+import com.pileproject.drivecommand.machine.device.port.OutputPort;
+
+public class NxtMachineSimulator extends NxtMachine {
```

Comment:
I think this class could targets not only NxtMachine but all classes deriving from MachineBase.
We can move this class to com.pileproject.drivecommand.mock.model, for example, and can also implement MockMachine inheriting MachineBase.